### ISSUE TITLE
Python 3.9 error: ImportError: cannot import name 'get_location' from 'textx.scoping.tools'

### ISSUE THREAD

**Issue by stanislaw** [textX/textX#312]
Kind greeting again from the developer of the StrictDoc project 😄 

[StrictDoc](https://github.com/strictdoc-project/strictdoc)'s CI passes green on Python 3.6, 3.7, and 3.8.

Now I am trying to enable Python 3.9 on the GitHub Actions CI for StrictDoc and I have started getting the following import error:

```
___________ ERROR collecting tests/unit/dsl/test_dsl_passthrough.py ____________
ImportError while importing test module '/Users/Stanislaw/workspace/projects/code/strictdoc/tests/unit/dsl/test_dsl_passthrough.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../../../.pyenv/versions/3.9.0/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/unit/dsl/test_dsl_passthrough.py:6: in <module>
    from strictdoc.backend.dsl.reader import SDReader
strictdoc/backend/dsl/reader.py:6: in <module>
    from textx.scoping.tools import get_location
E   ImportError: cannot import name 'get_location' from 'textx.scoping.tools' (/Users/Stanislaw/.pyenv/versions/3.9.0/lib/python3.9/site-packages/textx/scoping/tools.py)
```

My current quick workaround is to do this: 

```
if sys.version_info >= (3, 9):
    from textx.model import get_location
else:
    from textx.scoping.tools import get_location
```

Could you explain why this change is sensitive to the Python version change to 3.9 and could you confirm that this is a good workaround or suggest a better one?

Thank you.

**Comment by stanislaw** [comment]
Actually, a better workaround is to catch the `ImportError`:

```py
try:
    from textx.model import get_location
except ImportError:
    try:
        from textx.scoping.tools import get_location
    except ImportError:
        raise NotImplementedError(
            "Cannot import get_location function from textX"
        )
```

What would you recommend?

**Comment by goto40** [comment]
As far as I see it is not a Python 3.9 issue, but a refactoring we did in 2.0.1.

CHANGELOG.md: "textx_isinstance from textx.scoping.tools made available in textx (#164, #157)"

Since we found the function better fits in the core module, we *moved* the function: https://github.com/textX/textX/commit/08872a4ca3c5c8a76ea86a58aa7f1bfd0547e441

We are sorry about that incompatibility. You will have to change the imports...

**Comment by stanislaw** [comment]
Ah, I see. The reason I started hitting this because I sometimes `pip install` outside of Poetry's environment where the textx version was pinned to an older one. So I was checking 3.9 and pip installed the newer version of textx.

Now it makes sense.
